virtual_rainforest/models/hydrology/hydrology_model.py

```diff
@@ -127,12 +130,27 @@ def from_config(
         canopy_layers = config["core"]["layers"]["canopy_layers"]
         initial_soil_moisture = config["hydrology"]["initial_soil_moisture"]
 
+        # Check if any constants have been supplied
+        if "hydrology" in config and "constants" in config["hydrology"]:
+            # Checks that constants is config are as expected
+            check_valid_constant_names(config, "hydrology", "HydroConsts")
+            # If an error isn't raised then generate the dataclass
+            constants = HydroConsts(**config["hydrology"]["constants"]["HydroConsts"])
+        else:
+            # If no constants are supplied then the defaults should be used
+            constants = HydroConsts()
+
```

Comment:
When we set up the `constants` registry, that should include the config path in the registry details. That should then allow us to have a shared general function in `from_config` methods:

constants = load_constants(config, HydroConsts)

and move all of this into that shared function. We can key the registry dictionary using the classes, I think.

Reply:
I guess this is something for a separate PR because it will need to be adjusted across all models?